### PR TITLE
chore(v3): local debug e2e

### PR DESCRIPTION
**Pull request by jayzhang**

### CODE REVIEW

File: packages/vscode-extension/src/debug/commonUtils.ts [hunk 1]
```diff
@@ -109,36 +109,75 @@ export async function getDebugConfig(
...
+      if (!config)
+        throw new UserError("GetConfigError", "Failed to get project config", "Extension");
+      if (isLocalSideloadingConfiguration) {
+        const envInfo = config.envInfos["local"];
+        if (!envInfo)
+          throw new UserError("EnvConfigNotExist", "Local Env config not exist", "Extension");
+        const appId = envInfo.state["fx-resource-appstudio"].teamsAppId as string;
+        return { appId: appId, env: "local" };
+      } else {
+        if (env === undefined) {
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/commonUtils.ts:126]: What if env is defined?

File: packages/vscode-extension/src/debug/commonUtils.ts [hunk 2]
```diff
@@ -109,36 +109,75 @@ export async function getDebugConfig(
...
+          const inputs = getSystemInputs();
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/commonUtils.ts:127]: `const inputs = getSystemInputs()` is already called at line 113. Could that `inputs` be reused?

File: packages/vscode-extension/src/debug/commonUtils.ts [hunk 3]
```diff
@@ -109,36 +109,75 @@ export async function getDebugConfig(
...
+          env = envRes.value;
+          if (!env)
+            throw new UserError(
+              "GetSelectedEnvError",
+              "Failed to get selected Env name",
+              "Extension"
+            );
+          const envInfo = config.envInfos[env];
+          if (!envInfo)
+            throw new UserError("EnvConfigNotExist", `Env '${env} ' config not exist`, "Extension");
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/commonUtils.ts:144]: Tiny format - an extra space in `'${env} '`

File: packages/vscode-extension/src/debug/commonUtils.ts [hunk 4]
```diff
@@ -109,36 +109,75 @@ export async function getDebugConfig(
...
-      const localSettings = await localEnvManager.getLocalSettings(ext.workspaceUri.fsPath);
-      return { appId: localSettings?.teamsApp?.teamsAppId as string };
+    if (isV3()) {
+      const inputs = getSystemInputs();
+      const getConfigRes = await core.getProjectConfigV3(inputs);
+      if (getConfigRes.isErr()) throw getConfigRes.error;
+      const config = getConfigRes.value;
+      if (!config)
+        throw new UserError("GetConfigError", "Failed to get project config", "Extension");
+      if (isLocalSideloadingConfiguration) {
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/commonUtils.ts:119]: Does this mean `localSettings` is retired when `isV3` enabled?
Then which flag takes effect first, `isV3` or `isConfigUnifyEnabled`? @xiaolang124

Comment (xiaolang124) [reply packages/vscode-extension/src/debug/commonUtils.ts:119]: In this part, V3 has been forced to read local env. For now, we need to set ```isConfigUnifyEnabled``` to true to let create project generate local env config.

Comment (jayzhang) [reply packages/vscode-extension/src/debug/commonUtils.ts:119]: When isV3 is enabled, I think there is no isConfigUnifyEnabled anymore.

Comment (swatDong) [reply packages/vscode-extension/src/debug/commonUtils.ts:119]: I still see `localSettings` in remaining code. Will that be retired by `isV3` or `isConfigUnifyEnabled`?

Comment (jayzhang) [reply packages/vscode-extension/src/debug/commonUtils.ts:119]: unify config is the first step,  v3 is the second step.

**Review by swatDong — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Review by hund030 — APPROVED** [review]

**Review by IvanJobs — APPROVED** [review]
approved for bot.

**Review by nliu-ms — APPROVED** [review]

**Pull request merged**